<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
  <head>
    <meta http-equiv="Content-type" content="text/html; charset=utf-8" />
    <meta http-equiv="Content-Language" content="en-us" />
    <title>Django Exceptions | Django documentation | Django</title>
    <meta name="ROBOTS" content="ALL" />
    <meta http-equiv="imagetoolbar" content="no" />
    <meta name="MSSmartTagsPreventParsing" content="true" />
    <meta name="Copyright" content="Django Software Foundation" />
    <meta name="keywords" content="Python, Django, framework, open-source" />
    <meta name="description" content="Django is a high-level Python Web framework that encourages rapid development and clean, pragmatic design." />
    
    <link href="../../../s/css/base.383e0bb3119d.css" rel="stylesheet" type="text/css" media="screen" />
    
    <link href="../../../s/css/print.ac134bbb8dfc.css" rel="stylesheet" type="text/css" media="print" />
    
  
  <link rel="stylesheet" href="../../../s/css/docs/docs.feec2ab82630.css" type="text/css" />
  <link rel="stylesheet" href="../../../s/css/pygments.0d57d48be058.css" type="text/css" />

  </head>
  <body id="documentation" class="default">
  <div id="container">
    <div id="header">
      <h1 id="logo"><a href="https://www.djangoproject.com/"><img src="../../../s/img/site/hdr_logo.b19c5e60269d.gif" alt="Django" /></a></h1>
      <ul id="nav-global">
        <li id="nav-homepage"><a href="https://www.djangoproject.com/">Home</a></li>
        <li id="nav-download"><a href="https://www.djangoproject.com/download/">Download</a></li>
        <li id="nav-documentation"><a href="../../../index.html">Documentation</a></li>
        <li id="nav-weblog"><a href="https://www.djangoproject.com/weblog/">Weblog</a></li>
        <li id="nav-community"><a href="https://www.djangoproject.com/community/">Community</a></li>
        <li id="nav-code"><a href="https://code.djangoproject.com/">Code</a></li>
      </ul>
    </div>
    <!-- END Header -->
    <div id="billboard">
  <h2><a href="../../../index.html">Django documentation</a></h2>
</div>
    <div id="columnwrap">
      
		<div id="content-main">
		


  
  
  <ul id="doc-versions">
      
        
          <li class="other">
          
            <a href="../../1.1/ref/exceptions.1">1.1</a>
          
          </li>
        
      
        
          <li class="other">
          
            <a href="../../1.2/ref/exceptions.1">1.2</a>
          
          </li>
        
      
        
          <li class="other">
          
            <a href="../../1.3/ref/exceptions.1">1.3</a>
          
          </li>
        
      
        
          <li class="other">
          
            <a href="../../1.4/ref/exceptions.1">1.4</a>
          
          </li>
        
      
        
      
        
          <li class="other">
          
            <a href="../../1.6/ref/exceptions.1">1.6</a>
          
          </li>
        
      
        
          <li class="other">
          
            <a href="../../dev/ref/exceptions.1">dev</a>
          
          </li>
        
      
      <li class="current"
          title="This document describes Django 1.5. Click on the links on the left to see other versions.">
        <span>Documentation version:
          <strong>1.5
          </strong>
        </span>
      </li>
  </ul>
  <div class="section" id="s-django-exceptions">
<span id="django-exceptions"></span><h1>Django Exceptions<a class="headerlink" href="exceptions.1#django-exceptions" title="Permalink to this headline">¶</a></h1>
<p>Django raises some Django specific exceptions as well as many standard
Python exceptions.</p>
<div class="section" id="s-module-django.core.exceptions">
<span id="s-django-specific-exceptions"></span><span id="module-django.core.exceptions"></span><span id="django-specific-exceptions"></span><h2>Django-specific Exceptions<a class="headerlink" href="exceptions.1#module-django.core.exceptions" title="Permalink to this headline">¶</a></h2>
<div class="section" id="s-objectdoesnotexist-and-doesnotexist">
<span id="objectdoesnotexist-and-doesnotexist"></span><h3>ObjectDoesNotExist and DoesNotExist<a class="headerlink" href="exceptions.1#objectdoesnotexist-and-doesnotexist" title="Permalink to this headline">¶</a></h3>
<dl class="exception">
<dt id="django.core.exceptions.DoesNotExist">
<em class="property">exception </em><tt class="descname">DoesNotExist</tt><a class="headerlink" href="exceptions.1#django.core.exceptions.DoesNotExist" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="exception">
<dt id="django.core.exceptions.ObjectDoesNotExist">
<em class="property">exception </em><tt class="descname">ObjectDoesNotExist</tt><a class="headerlink" href="exceptions.1#django.core.exceptions.ObjectDoesNotExist" title="Permalink to this definition">¶</a></dt>
<dd><p>The <a class="reference internal" href="exceptions.1#django.core.exceptions.DoesNotExist" title="django.core.exceptions.DoesNotExist"><tt class="xref py py-exc docutils literal"><span class="pre">DoesNotExist</span></tt></a> exception is raised when an object is not found
for the given parameters of a query.</p>
<p><a class="reference internal" href="exceptions.1#django.core.exceptions.ObjectDoesNotExist" title="django.core.exceptions.ObjectDoesNotExist"><tt class="xref py py-exc docutils literal"><span class="pre">ObjectDoesNotExist</span></tt></a> is defined in <a class="reference internal" href="exceptions.1#module-django.core.exceptions" title="django.core.exceptions: Django specific exceptions"><tt class="xref py py-mod docutils literal"><span class="pre">django.core.exceptions</span></tt></a>.
<a class="reference internal" href="exceptions.1#django.core.exceptions.DoesNotExist" title="django.core.exceptions.DoesNotExist"><tt class="xref py py-exc docutils literal"><span class="pre">DoesNotExist</span></tt></a> is a subclass of the base <a class="reference internal" href="exceptions.1#django.core.exceptions.ObjectDoesNotExist" title="django.core.exceptions.ObjectDoesNotExist"><tt class="xref py py-exc docutils literal"><span class="pre">ObjectDoesNotExist</span></tt></a>
exception that is provided on every model class as a way of
identifying the specific type of object that could not be found.</p>
<p>See <a class="reference internal" href="models/querysets/index.html#django.db.models.query.QuerySet.get" title="django.db.models.query.QuerySet.get"><tt class="xref py py-meth docutils literal"><span class="pre">get()</span></tt></a> for further information
on <a class="reference internal" href="exceptions.1#django.core.exceptions.ObjectDoesNotExist" title="django.core.exceptions.ObjectDoesNotExist"><tt class="xref py py-exc docutils literal"><span class="pre">ObjectDoesNotExist</span></tt></a> and <a class="reference internal" href="exceptions.1#django.core.exceptions.DoesNotExist" title="django.core.exceptions.DoesNotExist"><tt class="xref py py-exc docutils literal"><span class="pre">DoesNotExist</span></tt></a>.</p>
</dd></dl>

</div>
<div class="section" id="s-multipleobjectsreturned">
<span id="multipleobjectsreturned"></span><h3>MultipleObjectsReturned<a class="headerlink" href="exceptions.1#multipleobjectsreturned" title="Permalink to this headline">¶</a></h3>
<dl class="exception">
<dt id="django.core.exceptions.MultipleObjectsReturned">
<em class="property">exception </em><tt class="descname">MultipleObjectsReturned</tt><a class="headerlink" href="exceptions.1#django.core.exceptions.MultipleObjectsReturned" title="Permalink to this definition">¶</a></dt>
<dd><p>The <a class="reference internal" href="exceptions.1#django.core.exceptions.MultipleObjectsReturned" title="django.core.exceptions.MultipleObjectsReturned"><tt class="xref py py-exc docutils literal"><span class="pre">MultipleObjectsReturned</span></tt></a> exception is raised by a query if only
one object is expected, but multiple objects are returned. A base version
of this exception is provided in <a class="reference internal" href="exceptions.1#module-django.core.exceptions" title="django.core.exceptions: Django specific exceptions"><tt class="xref py py-mod docutils literal"><span class="pre">django.core.exceptions</span></tt></a>; each model
class contains a subclassed version that can be used to identify the
specific object type that has returned multiple objects.</p>
<p>See <a class="reference internal" href="models/querysets/index.html#django.db.models.query.QuerySet.get" title="django.db.models.query.QuerySet.get"><tt class="xref py py-meth docutils literal"><span class="pre">get()</span></tt></a> for further information.</p>
</dd></dl>

</div>
<div class="section" id="s-suspiciousoperation">
<span id="suspiciousoperation"></span><h3>SuspiciousOperation<a class="headerlink" href="exceptions.1#suspiciousoperation" title="Permalink to this headline">¶</a></h3>
<dl class="exception">
<dt id="django.core.exceptions.SuspiciousOperation">
<em class="property">exception </em><tt class="descname">SuspiciousOperation</tt><a class="headerlink" href="exceptions.1#django.core.exceptions.SuspiciousOperation" title="Permalink to this definition">¶</a></dt>
<dd><p>The <a class="reference internal" href="exceptions.1#django.core.exceptions.SuspiciousOperation" title="django.core.exceptions.SuspiciousOperation"><tt class="xref py py-exc docutils literal"><span class="pre">SuspiciousOperation</span></tt></a> exception is raised when a user has performed
an operation that should be considered suspicious from a security perspective,
such as tampering with a session cookie.</p>
</dd></dl>

</div>
<div class="section" id="s-permissiondenied">
<span id="permissiondenied"></span><h3>PermissionDenied<a class="headerlink" href="exceptions.1#permissiondenied" title="Permalink to this headline">¶</a></h3>
<dl class="exception">
<dt id="django.core.exceptions.PermissionDenied">
<em class="property">exception </em><tt class="descname">PermissionDenied</tt><a class="headerlink" href="exceptions.1#django.core.exceptions.PermissionDenied" title="Permalink to this definition">¶</a></dt>
<dd><p>The <a class="reference internal" href="exceptions.1#django.core.exceptions.PermissionDenied" title="django.core.exceptions.PermissionDenied"><tt class="xref py py-exc docutils literal"><span class="pre">PermissionDenied</span></tt></a> exception is raised when a user does not have
permission to perform the action requested.</p>
</dd></dl>

</div>
<div class="section" id="s-viewdoesnotexist">
<span id="viewdoesnotexist"></span><h3>ViewDoesNotExist<a class="headerlink" href="exceptions.1#viewdoesnotexist" title="Permalink to this headline">¶</a></h3>
<dl class="exception">
<dt id="django.core.exceptions.ViewDoesNotExist">
<em class="property">exception </em><tt class="descname">ViewDoesNotExist</tt><a class="headerlink" href="exceptions.1#django.core.exceptions.ViewDoesNotExist" title="Permalink to this definition">¶</a></dt>
<dd><p>The <a class="reference internal" href="exceptions.1#django.core.exceptions.ViewDoesNotExist" title="django.core.exceptions.ViewDoesNotExist"><tt class="xref py py-exc docutils literal"><span class="pre">ViewDoesNotExist</span></tt></a> exception is raised by
<a class="reference internal" href="urlresolvers.1#module-django.core.urlresolvers" title="django.core.urlresolvers"><tt class="xref py py-mod docutils literal"><span class="pre">django.core.urlresolvers</span></tt></a> when a requested view does not exist.</p>
</dd></dl>

</div>
<div class="section" id="s-middlewarenotused">
<span id="middlewarenotused"></span><h3>MiddlewareNotUsed<a class="headerlink" href="exceptions.1#middlewarenotused" title="Permalink to this headline">¶</a></h3>
<dl class="exception">
<dt id="django.core.exceptions.MiddlewareNotUsed">
<em class="property">exception </em><tt class="descname">MiddlewareNotUsed</tt><a class="headerlink" href="exceptions.1#django.core.exceptions.MiddlewareNotUsed" title="Permalink to this definition">¶</a></dt>
<dd><p>The <a class="reference internal" href="exceptions.1#django.core.exceptions.MiddlewareNotUsed" title="django.core.exceptions.MiddlewareNotUsed"><tt class="xref py py-exc docutils literal"><span class="pre">MiddlewareNotUsed</span></tt></a> exception is raised when a middleware is not
used in the server configuration.</p>
</dd></dl>

</div>
<div class="section" id="s-improperlyconfigured">
<span id="improperlyconfigured"></span><h3>ImproperlyConfigured<a class="headerlink" href="exceptions.1#improperlyconfigured" title="Permalink to this headline">¶</a></h3>
<dl class="exception">
<dt id="django.core.exceptions.ImproperlyConfigured">
<em class="property">exception </em><tt class="descname">ImproperlyConfigured</tt><a class="headerlink" href="exceptions.1#django.core.exceptions.ImproperlyConfigured" title="Permalink to this definition">¶</a></dt>
<dd><p>The <a class="reference internal" href="exceptions.1#django.core.exceptions.ImproperlyConfigured" title="django.core.exceptions.ImproperlyConfigured"><tt class="xref py py-exc docutils literal"><span class="pre">ImproperlyConfigured</span></tt></a> exception is raised when Django is
somehow improperly configured &#8211; for example, if a value in <tt class="docutils literal"><span class="pre">settings.py</span></tt>
is incorrect or unparseable.</p>
</dd></dl>

</div>
<div class="section" id="s-fielderror">
<span id="fielderror"></span><h3>FieldError<a class="headerlink" href="exceptions.1#fielderror" title="Permalink to this headline">¶</a></h3>
<dl class="exception">
<dt id="django.core.exceptions.FieldError">
<em class="property">exception </em><tt class="descname">FieldError</tt><a class="headerlink" href="exceptions.1#django.core.exceptions.FieldError" title="Permalink to this definition">¶</a></dt>
<dd><p>The <a class="reference internal" href="exceptions.1#django.core.exceptions.FieldError" title="django.core.exceptions.FieldError"><tt class="xref py py-exc docutils literal"><span class="pre">FieldError</span></tt></a> exception is raised when there is a problem with a
model field. This can happen for several reasons:</p>
<ul class="simple">
<li>A field in a model clashes with a field of the same name from an
abstract base class</li>
<li>An infinite loop is caused by ordering</li>
<li>A keyword cannot be parsed from the filter parameters</li>
<li>A field cannot be determined from a keyword in the query
parameters</li>
<li>A join is not permitted on the specified field</li>
<li>A field name is invalid</li>
<li>A query contains invalid order_by arguments</li>
</ul>
</dd></dl>

</div>
<div class="section" id="s-validationerror">
<span id="validationerror"></span><h3>ValidationError<a class="headerlink" href="exceptions.1#validationerror" title="Permalink to this headline">¶</a></h3>
<dl class="exception">
<dt id="django.core.exceptions.ValidationError">
<em class="property">exception </em><tt class="descname">ValidationError</tt><a class="headerlink" href="exceptions.1#django.core.exceptions.ValidationError" title="Permalink to this definition">¶</a></dt>
<dd><p>The <a class="reference internal" href="exceptions.1#django.core.exceptions.ValidationError" title="django.core.exceptions.ValidationError"><tt class="xref py py-exc docutils literal"><span class="pre">ValidationError</span></tt></a> exception is raised when data fails form or
model field validation. For more information about validation, see
<a class="reference internal" href="forms/validation/index.html"><em>Form and Field Validation</em></a>,
<a class="reference internal" href="models/instances/index.html#validating-objects"><em>Model Field Validation</em></a> and the
<a class="reference internal" href="validators.1"><em>Validator Reference</em></a>.</p>
</dd></dl>

</div>
<div class="section" id="s-noreversematch">
<span id="noreversematch"></span><h3>NoReverseMatch<a class="headerlink" href="exceptions.1#noreversematch" title="Permalink to this headline">¶</a></h3>
<dl class="exception">
<dt id="django.core.urlresolvers.NoReverseMatch">
<em class="property">exception </em><tt class="descname">NoReverseMatch</tt><a class="headerlink" href="exceptions.1#django.core.urlresolvers.NoReverseMatch" title="Permalink to this definition">¶</a></dt>
<dd><p>The <a class="reference internal" href="exceptions.1#django.core.urlresolvers.NoReverseMatch" title="django.core.urlresolvers.NoReverseMatch"><tt class="xref py py-exc docutils literal"><span class="pre">NoReverseMatch</span></tt></a> exception is raised by
<a class="reference internal" href="urlresolvers.1#module-django.core.urlresolvers" title="django.core.urlresolvers"><tt class="xref py py-mod docutils literal"><span class="pre">django.core.urlresolvers</span></tt></a> when a matching URL in your URLconf
cannot be identified based on the parameters supplied.</p>
</dd></dl>

</div>
</div>
<div class="section" id="s-database-exceptions">
<span id="database-exceptions"></span><h2>Database Exceptions<a class="headerlink" href="exceptions.1#database-exceptions" title="Permalink to this headline">¶</a></h2>
<p>Django wraps the standard database exceptions <a class="reference internal" href="exceptions.1#django.db.DatabaseError" title="django.db.DatabaseError"><tt class="xref py py-exc docutils literal"><span class="pre">DatabaseError</span></tt></a> and
<a class="reference internal" href="exceptions.1#django.db.IntegrityError" title="django.db.IntegrityError"><tt class="xref py py-exc docutils literal"><span class="pre">IntegrityError</span></tt></a> so that your Django code has a guaranteed common
implementation of these classes. These database exceptions are
provided in <a class="reference internal" href="../topics/db/index#module-django.db" title="django.db"><tt class="xref py py-mod docutils literal"><span class="pre">django.db</span></tt></a>.</p>
<dl class="exception">
<dt id="django.db.DatabaseError">
<em class="property">exception </em><tt class="descname">DatabaseError</tt><a class="headerlink" href="exceptions.1#django.db.DatabaseError" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="exception">
<dt id="django.db.IntegrityError">
<em class="property">exception </em><tt class="descname">IntegrityError</tt><a class="headerlink" href="exceptions.1#django.db.IntegrityError" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<p>The Django wrappers for database exceptions behave exactly the same as
the underlying database exceptions. See <span class="target" id="index-0"></span><a class="pep reference external" href="http://www.python.org/dev/peps/pep-0249"><strong>PEP 249</strong></a>, the Python Database API
Specification v2.0, for further information.</p>
<dl class="exception">
<dt id="django.db.models.ProtectedError">
<em class="property">exception </em><tt class="descclassname">models.</tt><tt class="descname">ProtectedError</tt><a class="headerlink" href="exceptions.1#django.db.models.ProtectedError" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<p>Raised to prevent deletion of referenced objects when using
<a class="reference internal" href="models/fields.1#django.db.models.PROTECT" title="django.db.models.PROTECT"><tt class="xref py py-attr docutils literal"><span class="pre">django.db.models.PROTECT</span></tt></a>. Subclass of <a class="reference internal" href="exceptions.1#django.db.IntegrityError" title="django.db.IntegrityError"><tt class="xref py py-exc docutils literal"><span class="pre">IntegrityError</span></tt></a>.</p>
</div>
<div class="section" id="s-http-exceptions">
<span id="http-exceptions"></span><h2>Http Exceptions<a class="headerlink" href="exceptions.1#http-exceptions" title="Permalink to this headline">¶</a></h2>
<dl class="exception">
<dt id="django.http.UnreadablePostError">
<em class="property">exception </em><tt class="descname">UnreadablePostError</tt><a class="headerlink" href="exceptions.1#django.http.UnreadablePostError" title="Permalink to this definition">¶</a></dt>
<dd><p>The <a class="reference internal" href="exceptions.1#django.http.UnreadablePostError" title="django.http.UnreadablePostError"><tt class="xref py py-exc docutils literal"><span class="pre">UnreadablePostError</span></tt></a> is raised when a user cancels an upload.
It is available from <a class="reference internal" href="request-response.1#module-django.http" title="django.http: Classes dealing with HTTP requests and responses."><tt class="xref py py-mod docutils literal"><span class="pre">django.http</span></tt></a>.</p>
</dd></dl>

</div>
<div class="section" id="s-transaction-exceptions">
<span id="transaction-exceptions"></span><h2>Transaction Exceptions<a class="headerlink" href="exceptions.1#transaction-exceptions" title="Permalink to this headline">¶</a></h2>
<dl class="exception">
<dt id="django.db.transaction.TransactionManagementError">
<em class="property">exception </em><tt class="descname">TransactionManagementError</tt><a class="headerlink" href="exceptions.1#django.db.transaction.TransactionManagementError" title="Permalink to this definition">¶</a></dt>
<dd><p>The <a class="reference internal" href="exceptions.1#django.db.transaction.TransactionManagementError" title="django.db.transaction.TransactionManagementError"><tt class="xref py py-exc docutils literal"><span class="pre">TransactionManagementError</span></tt></a> is raised for any and all problems
related to database transactions. It is available from
<a class="reference internal" href="../topics/db/transactions.1#module-django.db.transaction" title="django.db.transaction"><tt class="xref py py-mod docutils literal"><span class="pre">django.db.transaction</span></tt></a>.</p>
</dd></dl>

</div>
<div class="section" id="s-python-exceptions">
<span id="python-exceptions"></span><h2>Python Exceptions<a class="headerlink" href="exceptions.1#python-exceptions" title="Permalink to this headline">¶</a></h2>
<p>Django raises built-in Python exceptions when appropriate as well. See the
Python documentation for further information on the
built-in <a class="reference external" href="http://docs.python.org/2.7/library/exceptions.html#module-exceptions" title="(in Python v2.7)"><tt class="xref py py-mod docutils literal"><span class="pre">exceptions</span></tt></a>.</p>
</div>
</div>


<div class="browse-horizontal">
  
    <div class="left"><a href="django-admin/index.html">&laquo; django-admin.py and manage.py</a></div>
  
  
    <div class="right"><a href="files/index">File handling &raquo;</a></div>
  
</div>
<div id="content-secondary">
  <h2 id="comments">Questions/Feedback</h2>
  <p>Having trouble? We'd like to help!</p>
  <ul>
    <li>
      
      Try the <a href="../faq/index.html">FAQ</a> — it's got answers to many common questions.
    </li>
    <li>
      Search for information in the <a
      href="http://groups.google.com/group/django-users/">archives of the
      django-users mailing list</a>, or <a
      href="http://groups.google.com/group/django-users/">post a question</a>.
    </li>
    <li>
      Ask a question in the <a href="irc://irc.freenode.net/">#django IRC
      channel</a>, or search the <a href="http://django-irc-logs.com/">IRC
      logs</a> to see if it has been asked before.
    </li>
    <li>
      If you notice errors with this documentation, please <a
      href="https://code.djangoproject.com/newticket?component=Documentation">
      open a ticket</a> and let us know! Please only use the ticket tracker for
      criticisms and improvements on the docs. For tech support, use the
      resources above.
    </li>
  </ul>
</div>

		</div>
		<!-- END #content-main -->
		<div id="content-related" class="sidebar">
		
  
    <h2>Search</h2>
    
    

<form action="https://docs.djangoproject.com/search/" id="sidebar_search" class="search">
  <div>
    <p>
      <input id="id_sidebar_search_q" name="q" type="search" />
    </p>
    <p>
      <label for="id_sidebar_search_release">Version:</label>
      <select id="id_sidebar_search_release" name="release">
<option value="2">Django 1.0</option>
<option value="3">Django 1.1</option>
<option value="4">Django 1.2</option>
<option value="5">Django 1.3</option>
<option value="6">Django 1.4</option>
<option value="7" selected="selected">Django 1.5</option>
<option value="9">Django 1.6</option>
<option value="1">Development trunk</option>
</select>
    </p>
    <p>
      <input type="submit" class="submit" value="Search" />
    </p>
  </div>
</form>

  

  
    <h2>Contents</h2>
    
      <ul>
<li><a class="reference internal" href="exceptions.1#">Django Exceptions</a><ul>
<li><a class="reference internal" href="exceptions.1#module-django.core.exceptions">Django-specific Exceptions</a><ul>
<li><a class="reference internal" href="exceptions.1#objectdoesnotexist-and-doesnotexist">ObjectDoesNotExist and DoesNotExist</a></li>
<li><a class="reference internal" href="exceptions.1#multipleobjectsreturned">MultipleObjectsReturned</a></li>
<li><a class="reference internal" href="exceptions.1#suspiciousoperation">SuspiciousOperation</a></li>
<li><a class="reference internal" href="exceptions.1#permissiondenied">PermissionDenied</a></li>
<li><a class="reference internal" href="exceptions.1#viewdoesnotexist">ViewDoesNotExist</a></li>
<li><a class="reference internal" href="exceptions.1#middlewarenotused">MiddlewareNotUsed</a></li>
<li><a class="reference internal" href="exceptions.1#improperlyconfigured">ImproperlyConfigured</a></li>
<li><a class="reference internal" href="exceptions.1#fielderror">FieldError</a></li>
<li><a class="reference internal" href="exceptions.1#validationerror">ValidationError</a></li>
<li><a class="reference internal" href="exceptions.1#noreversematch">NoReverseMatch</a></li>
</ul>
</li>
<li><a class="reference internal" href="exceptions.1#database-exceptions">Database Exceptions</a></li>
<li><a class="reference internal" href="exceptions.1#http-exceptions">Http Exceptions</a></li>
<li><a class="reference internal" href="exceptions.1#transaction-exceptions">Transaction Exceptions</a></li>
<li><a class="reference internal" href="exceptions.1#python-exceptions">Python Exceptions</a></li>
</ul>
</li>
</ul>

    
  
  
    <h2>Browse</h2>
    <ul>
      
        
          <li>Prev: <a href="django-admin/index.html">django-admin.py and manage.py</a></li>
        
        
          <li>Next: <a href="files/index">File handling</a></li>
        
        <li><a href="../contents/index.html">Table of contents</a></li>
        
          <li><a href="../genindex/index.html">General Index</a></li>
        
          <li><a href="../py-modindex/index.html">Python Module Index</a></li>
        
      
    </ul>
  

  
    <h2>You are here:</h2>
    <ul>
      
        <li>
          <a href="../../../index.html">Django 1.5 documentation</a>
          
            <ul><li><a href="index.html">API Reference</a>
          
          <ul><li>Django Exceptions</li></ul>
          </li></ul>
        </li>
      
    </ul>

    
      <h2>Download:</h2>
      <p>
        Offline (Django 1.5):
        <a href="../../../m/docs/django-docs-1.5-en.zip">HTML</a> |
        <a href="http://media.readthedocs.org/pdf/django/1.5.x/django.pdf">PDF</a> |
        <a href="http://media.readthedocs.org/epub/django/1.5.x/django.epub">ePub</a>
      <br>
      <span class="quiet">
        Provided by <a href="http://readthedocs.org/">Read the Docs</a>.
      </span>
      </p>
    
  

		</div>
		<!-- END #content-related -->

    </div>
    <!-- END #content -->
    <div id="footer">
      <p>&copy; 2005-2013
        <a href="https://www.djangoproject.com/foundation/"> Django Software
        Foundation</a> unless otherwise noted. Django is a
        <a href="https://www.djangoproject.com/trademarks/">registered
        trademark</a> of the Django Software Foundation.
        <a href="http://mediatemple.net/">Linux Web hosting</a> graciously
        provided by Media Temple.
      </p>
    </div>
    <!-- END #footer -->
  </div>
  <!-- END #container -->
  


  </body>
</html>
